pkg/metadata/annotation_management_mode.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"kpt.dev/configsync/pkg/core"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ManagementMode is the type used to identify value enums to use with the
+// `configmanagement.gke.io/managed` annotation.
+type ManagementMode string
+
+// String returns the string value of the ManagementMode.
+// Implements the Stringer interface.
+func (m ManagementMode) String() string {
+	return string(m)
+}
+
+const (
+	// ManagementModeAnnotationKey is the annotation that indicates whether
+	// Config Sync should manage the content and lifecycle for the object.
+	// This annotation is set by Config Sync on a managed resource object.
+	ManagementModeAnnotationKey = ConfigManagementPrefix + "managed"
+	// ManagementEnabled is the value corresponding to
+	// ManagementModeAnnotationKey indicating that Config Sync should manage
+	// content and lifecycle for the given resource object.
+	ManagementEnabled ManagementMode = "enabled"
+	// ManagementDisabled is the value corresponding to
+	// ManagementModeAnnotationKey indicating that Config Sync should not manage
+	// content and lifecycle for the given resource.
+	//
+	// By design, the `configmanagement.gke.io/managed: disabled` annotation
+	// should not be pushed to the cluster. Instead, we remove all the Config
+	// Sync metadata from the object on the cluster.
+	ManagementDisabled ManagementMode = "disabled"
+)
+
+// WithManagementMode returns a MetaMutator that sets the managed annotation on an Object.
+func WithManagementMode(mode ManagementMode) core.MetaMutator {
+	return core.Annotation(ManagementModeAnnotationKey, mode.String())
+}
+
+// WithoutManagementMode returns a MetaMutator that removes the managed annotation on an Object.
+func WithoutManagementMode() core.MetaMutator {
+	return core.WithoutAnnotation(ManagementModeAnnotationKey)
+}
+
+// IsManagementEnabled returns true if the object has the annotation
+// `configmanagement.gke.io/managed: enabled`.
+func IsManagementEnabled(obj client.Object) bool {
+	return core.GetAnnotation(obj, ManagementModeAnnotationKey) == ManagementEnabled.String()
+}
+
+// IsManagementDisabled returns true if the object has the annotation
+// `configmanagement.gke.io/managed: disabled`.
+func IsManagementDisabled(obj client.Object) bool {
```

Comment:
Should this default to true if the annotation is missing?

Reply:
Depends on usage. I initially had a different GetManagementMode that made that assumption and found it confusing to use, so I removed it. It seems like most of the places that test for disabled were really checking for explicitly disabled. So I figured it was better to have these checks be explicit.

Reply:
Ack, after seeing more of the code I think this makes sense